jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -958,6 +996,46 @@ public String toString()
         {
             return String.format("%s[i=%s,cb=%s]", super.toString(), _info, _callback);
         }
+
+        private static class CancelSendException extends IOException
+        {
+            private final CountDownLatch _complete = new CountDownLatch(1);
+            private Callback _callback;
+
+            public CancelSendException(Throwable cause)
+            {
+                super(cause);
+            }
+
+            public void complete()
+            {
+                _complete.countDown();
+            }
+
+            public void join()
```

Comment:
Make `join` return the `Callback` and get rid of `getCallback`?

Reply:
I don't like this.... @sbordet, what are your thoughts?

Reply:
That's what it does already, except returning null is more efficient than returning NOOP as it doesn't need to be combined with other callbacks (hmmm we can probably make combine handle that better).